pkg/localstore/mode_put.go

```diff
@@ -259,21 +262,31 @@ func (db *DB) putSync(batch *leveldb.Batch, binIDs map[uint8]uint64, item shed.I
 	if err != nil {
 		return false, 0, err
 	}
-	gcSizeChange, err = db.setGC(batch, item)
+	err = db.postageChunksIndex.PutInBatch(batch, item)
+	if err != nil {
+		return false, 0, err
+	}
+	gcSizeChange, err = db.preserveOrCache(batch, item, false)
 	if err != nil {
 		return false, 0, err
 	}
 
 	return false, gcSizeChange, nil
 }
 
-// setGC is a helper function used to add chunks to the retrieval access
-// index and the gc index in the cases that the putToGCCheck condition
-// warrants a gc set. this is to mitigate index leakage in edge cases where
-// a chunk is added to a node's localstore and given that the chunk is
-// already within that node's NN (thus, it can be added to the gc index
-// safely)
-func (db *DB) setGC(batch *leveldb.Batch, item shed.Item) (gcSizeChange int64, err error) {
+// preserveOrCache is a helper function used to add chunks to either a pinned reserve or gc cache
+// (the retrieval access index and the gc index)
+func (db *DB) preserveOrCache(batch *leveldb.Batch, item shed.Item, forcePin bool) (gcSizeChange int64, err error) {
+	// item needs to be populated with Radius
+	item2, err := db.postageRadiusIndex.Get(item)
+	if err != nil {
+		return 0, fmt.Errorf("postage radius index: %w", err)
+	}
+	item.Radius = item2.Radius
+
+	if withinRadiusFn(db, item) || forcePin {
```

Comment:
when this is from a simple setsync call we also should not set the access timestamp to now.
only if it is putRequest.

but surely not ike this, instead:
*make access update consistent irrespective of reserve/cache status*. (as reserve can be evicted after a second and then will be deleted even before their earlier cached (and earlier accessed) sister chunks.